source/val/validate.cpp

```diff
@@ -337,6 +315,32 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
   // for() above as it loops over all ordered_instructions internally.
   if (auto error = ValidateBuiltIns(*vstate)) return error;
 
+  // Validate individual opcodes.
+  for (size_t i = 0; i < vstate->ordered_instructions().size(); ++i) {
```

Comment:
There are other things that are going to be missing like in_function_body() and in_block() I believe. Can we add a flag to validation state that those are available, set it false after the first for() exist and then assert if any of those special methods are hit in later passes?

Reply:
Makes sense.

Reply:
current_function() and in_function_body() look like they get set correctly. current_function() asserts that in_function_body() is true. Likewise, in_block() looks ok too.